### DISCUSSION TITLE
Has Next started using inline styles in production builds?

### DISCUSSION THREAD

**Discussion by sophiekoonin** [vercel/next.js#18395] [Help]
We have quite a restrictive Content-Security-Policy (CSP) which blocks `unsafe-inline` as a matter of caution. It's always been fine with our Next setup (we use CSS modules and host our assets on a CDN). I just updated to Next 10 and now on first load I'm getting an unstyled page, because our CSP is refusing to load the unsafe styles. Looking in the DOM I can see a load of CSS in `<style>` tags that never used to be there. Is there a way of disabling this so we don't have to enable `unsafe-inline`?

**Comment by Timer** [comment]
There's currently no way to disable the inline styles, and we're exploring on relying on them even heavier in the future for performance optimizations. Could you open a Next.js issue to discuss this further, though?

**Reply by sophiekoonin** [reply]
https://github.com/vercel/next.js/issues/18451 ✨

**Reply by sophiekoonin** [reply]
hey @Timer  - I created an issue but it seems to have been converted into a discussion again!

**Reply by Timer** [reply]
That's odd—someone in our community moderators might've done it by accident!

**Reply by sophiekoonin** [reply]
Is it possible to convert it back? This is blocking Monzo from upgrading :(

**Reply by Timer** [reply]
Seems GitHub does not allow converting back 😞 Can you open it again and I can immediately tag it with stuff that should prevent someone from converting it?

**Reply by Timer** [reply]
Or I can open it by C+Ping your discussion?

**Reply by sophiekoonin** [reply]
Sure, I will do!

**Reply by sophiekoonin** [reply]
Not able to tag it myself, but: https://github.com/vercel/next.js/issues/18557

**Reply by Timer** [reply]
Got it tagged! It should stay as an issue now.
